source/dbo/Stored Procedures/Power BI Reporting/uspGenerateInitialSputumResults.sql

```diff
@@ -0,0 +1,79 @@
+/*
+Populate the InitialSputumSmearResult and InitialSputumPCRResult fields for each case
+Note: in all cases ‘sputum’ includes both sample types sputum - induced and sputum - spontaneous
+Calculated as follows for both smear and pcr:
+	If (any Positive or Negative results): 
+		Find results r from earliest date
+		if (r only contains negative results):
+			'Negative'
+		else:
+			'Positive'
+	Else if (any results are 'No result available'):
+		'No Result available'
+	Else if (any results 'Awaiting'):
+		'Awaiting'
+	Else (no results):
+		'No result'
+*/
+
+
+CREATE PROCEDURE [dbo].[uspGenerateInitialSputumResults]
+
+AS
+	WITH
+
+	ResultRanking AS
+	(
+		SELECT 1 AS [Rank], 1 AS [SubRank], 'Positive' AS ResultName
+		UNION
+		SELECT 1 AS [Rank], 2 AS [SubRank], 'Negative' AS ResultName
+		UNION
+		SELECT 2 AS [Rank], NULL AS [SubRank], 'NoResultAvailable' AS ResultName
+		UNION
+		SELECT 3 AS [Rank], NULL AS [SubRank], 'Awaiting' AS ResultName
```

Comment:
Very minor but seems weird to have some of the column names in brackets and some not